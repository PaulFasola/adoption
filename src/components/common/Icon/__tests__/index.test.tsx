import React from 'react';
import toJson from "enzyme-to-json";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, shallow } from 'enzyme';
import { Icon, IconType } from "..";

configure({ adapter: new Adapter() });

describe(`Icon`, () => {
  it('Should render', () => {
    Object.keys(IconType).forEach((key) => shallow(<Icon type={IconType[key]} />));
  });

  it('Should be clickable when url is defined', () => {
    const wrapper = shallow(<Icon type={IconType.OutboundLink} url="https://brrr.money/" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should be clickable when url is defined and the target should be _blank', () => {
    const wrapper = shallow(<Icon type={IconType.OutboundLink} url="https://brrr.money/" targetBlank />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});