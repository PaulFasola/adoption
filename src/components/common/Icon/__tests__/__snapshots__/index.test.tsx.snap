// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Should be clickable when url is defined 1`] = `
<styled.a
  href="https://brrr.money/"
  title="https://brrr.money/"
>
  <styled.div>
    <svg
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
      />
      <polyline
        points="15 3 21 3 21 9"
      />
      <line
        x1="10"
        x2="21"
        y1="14"
        y2="3"
      />
    </svg>
  </styled.div>
</styled.a>
`;

exports[`Icon Should be clickable when url is defined and the target should be _blank 1`] = `
<styled.a
  href="https://brrr.money/"
  rel="noreferrer"
  target="_blank"
  title="https://brrr.money/"
>
  <styled.div>
    <svg
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
      />
      <polyline
        points="15 3 21 3 21 9"
      />
      <line
        x1="10"
        x2="21"
        y1="14"
        y2="3"
      />
    </svg>
  </styled.div>
</styled.a>
`;
